### PR TITLE
deprecating envoy.reloadable_features.deprecate_global_int

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: inline
Fixes https://github.com/envoyproxy/envoy/issues/23593

### CODE REVIEW

File: source/common/http/header_map_impl.cc [hunk 1]
```diff
@@ -22,6 +22,7 @@ namespace {
 
 constexpr absl::string_view DelimiterForInlineHeaders{","};
 constexpr absl::string_view DelimiterForInlineCookies{"; "};
+const static int kMinHeadersForLazyMap = 3; // Arbitrary hard-coded value.
```
Comment (jmarantz) [inline source/common/http/header_map_impl.cc:25]: I think it's better than arbitrary :) @adisuissa did ran microbenchmarks to at least sanity-check that choice.

File: source/common/http/header_map_impl.h [hunk 2]
```diff
@@ -181,10 +181,9 @@ class HeaderMapImpl : NonCopyable {
   /**
    * List of HeaderEntryImpl that keeps the pseudo headers (key starting with ':') in the front
    * of the list (as required by nghttp2) and otherwise maintains insertion order.
-   * When the list size is greater or equal to the envoy.http.headermap.lazy_map_min_size runtime
-   * feature value (defaults to 3, if not set), all headers are added to a map, to allow
-   * fast access given a header key. Once the map is initialized, it will be used even if the number
-   * of headers decreases below the threshold.
+   * When the list size is greater or equal to the default min size (3), all headers are added to a
```
Comment (jmarantz) [inline source/common/http/header_map_impl.h:184]: s/default// ?

**Review by jmarantz — COMMENTED** [review]
basically lgtm; makes sense to me to drop that settings as no one ever really changed it to my knowledge :)

Is there some general reason not to support int settings though? It seems this PR is part of a bigger mission I was not aware of.

File: source/common/http/header_map_impl.cc (hunk 1 above)
Comment (adisuissa) [reply source/common/http/header_map_impl.cc:25]: 3 was chosen after doing some benchmarks (#12656).

**Review by jmarantz — APPROVED** [review]
thanks!

**Pull request merged**